uaclient/clouds/identity.py

```diff
@@ -75,7 +77,11 @@ def cloud_instance_factory() -> clouds.AutoAttachCloudInstance:
         "gce": gcp.UAAutoAttachGCPInstance,
     }  # type: Dict[str, Type[clouds.AutoAttachCloudInstance]]
 
-    cloud_type, _ = get_cloud_type()
+    if cloud_override is not None:
```

Comment:
Should we check if the `cloud_override` is valid ?

Reply:
That happens in the following lines in the same way that the output of cloud-id is checked. We look up the value in the dictionary of cloud-name -> cloud auto-attach class. If it isn't there, then we say we raise an error saying we don't support this cloud. Let me know if you think that is not sufficient